### PR TITLE
feat(webpack): move web-specific options into withWeb plugin

### PR DESCRIPTION
**Pull request by jaysoo**
This PR moves web-specific options from `@nrwl/webpack:webpack` executor to the `withWeb` and `withReact` plugins. 

For example, 

```js
const { composePlugins, withNx } = require('@nrwl/webpack');
const { withReact } = require('@nrwl/react');

// Nx plugins for webpack.
module.exports = composePlugins(withNx(), withReact({ subresourceIntegrity: true }), (config) => {
  // Update the webpack config as needed here.
  // e.g. `config.plugins.push(new MyPlugin())`
  return config;
});
```

## TODO

- [ ] Wait for https://github.com/nrwl/nx/pull/14618 to be merged since there is no other way to detect whether to apply Nx plugins or rely on users to apply them.

### CODE REVIEW

File: packages/webpack/package.json [hunk 1]
```diff
@@ -50,7 +50,6 @@
     "loader-utils": "^2.0.3",
     "mini-css-extract-plugin": "~2.4.7",
     "parse5": "4.0.0",
-    "parse5-html-rewriting-stream": "6.0.1",
```
Comment (jaysoo) [inline packages/webpack/package.json:53]: This is no longer needed now that both `index.html` plugins have beeb consolidated into `WriteIndexHtmlPlugin`. Previously we had two different ones for `dev-server` and `build`.

File: packages/webpack/src/executors/dev-server/dev-server.impl.ts [hunk 2]
```diff
@@ -83,7 +82,6 @@ export async function* devServerExecutor(
       map(({ baseUrl, stats }) => {
         return {
           baseUrl,
-          emittedFiles: getEmittedFiles(stats),
```
Comment (jaysoo) [inline packages/webpack/src/executors/dev-server/dev-server.impl.ts:86]: This is not used at all, and excluding it removes the need for the `getEmittedFiles` function in `lib`.

File: packages/webpack/src/executors/dev-server/lib/get-dev-server-config.ts [hunk 3]
```diff
@@ -28,27 +25,6 @@ export function getDevServerConfig(
...
-  const {
-    deployUrl,
-    subresourceIntegrity,
-    scripts = [],
-    styles = [],
-    index,
-    baseHref,
-  } = buildOptions;
-
-  webpackConfig.plugins.push(
```
Comment (jaysoo) [inline packages/webpack/src/executors/dev-server/lib/get-dev-server-config.ts:40]: This plugin is now in `withWeb` as `WriteIndexHtmlPlugin`.

File: packages/webpack/src/executors/webpack/schema.d.ts [hunk 4]
```diff
@@ -64,22 +57,40 @@ export interface WebpackExecutorOptions {
...
-  styles?: Array<ExtraEntryPointClass | string>;
-  subresourceIntegrity?: boolean;
   target?: 'node' | 'web';
   transformers?: TransformerEntry[];
   tsConfig: string;
   vendorChunk?: boolean;
   verbose?: boolean;
   watch?: boolean;
   webpackConfig?: string;
+  // TODO(jack): Also deprecate these in schema.json once we have migration from executor options to webpack.config.js file.
```
Comment (jaysoo) [inline packages/webpack/src/executors/webpack/schema.d.ts:71]: These are just reordered with new `deprecated comments.

Comment (ndcunningham) [reply packages/webpack/src/executors/webpack/schema.d.ts:71]: Is there a way to 'deep link' the deprecated message to the new function?

**Review by ndcunningham — APPROVED** [review]

**Pull request merged**